scripts/test/lld.py

```diff
@@ -49,10 +49,24 @@ def run_test(input_path):
             else:
                 continue
 
-        cmd = shared.WASM_EMSCRIPTEN_FINALIZE + [input_path, '-S'] + args
-        cmd += args_for_finalize(os.path.basename(input_path))
+        lowered_input_path = input_path
+        cmd = shared.WASM_EMSCRIPTEN_FINALIZE + args
+        if '64' in input_path:
+            # We're processing a wasm64 module, which normally gets lowered
+            # by emcc -S MEMORY64=2, so here we have to do that manually.
+            lowered_input_path = input_path + '.wasm32'
+            support.run_command(shared.WASM_OPT +
+                                ['--enable-memory64', '--memory64-lowering',
+                                 '-S', '-o', lowered_input_path, input_path])
+            cmd += ['--enable-memory64', '--bigint']
```

Comment:
But wait.. in emscripten we run the lowering pass *after* finalize.    So finalize should be designed to run on the non-lowered version right? 

wasm-emscripten-finalize is designed to take the direct output of lld as its input.   If you need the lowering to happen first it should be done in wasm-emscripten-finalize itself I think.

Reply:
It may not be needed for the current tests. I added it because it appeared that some code would not be able to deal with the full wasm64, and instead would need the lowered version.

I can remove it for now.